src/Compilers/CSharp/Portable/Compiler/MethodCompiler.cs

```diff
@@ -221,13 +221,42 @@ private static MethodSymbol GetEntryPoint(CSharpCompilation compilation, PEModul
                 !hasDeclarationErrors &&
                 !diagnostics.HasAnyErrors())
             {
-                var body = synthesizedEntryPoint.CreateBody();
+                BoundStatement body = synthesizedEntryPoint.CreateBody();
+
+                var dynamicAnalysisSpans = ImmutableArray<SourceSpan>.Empty;
+                var diagsForCurrentMethod = DiagnosticBag.GetInstance();
```

Comment:
Any chance of errors in lowering? If so, please test that case. Otherwise assert `diagsForCurrentMethod` is empty.